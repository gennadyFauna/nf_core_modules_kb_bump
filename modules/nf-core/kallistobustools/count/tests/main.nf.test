nextflow_process {

    name "Test Process KALLISTOBUSTOOLS_COUNT"
    script "../main.nf"
    process "KALLISTOBUSTOOLS_COUNT"

    tag "modules"
    tag "modules_nfcore"
    tag "kallistobustools"
    tag "kallistobustools/count"

    setup {
        run("KALLISTOBUSTOOLS_REF") {
            script "../../ref/main.nf"
            process {
                """
                input[0] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)
                input[2] = "standard"
                """
            }
        }
    }

    test("genome.fasta + genome.gtf + '10X3' + 'standard'") {

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [id:'test'], // meta map
                        [
                        file(params.test_data['homo_sapiens']['10xgenomics']['cellranger']['test_10x_10k_pbmc_cmo_gex1_fastq_1_gz'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['10xgenomics']['cellranger']['test_10x_10k_pbmc_cmo_gex1_fastq_2_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = KALLISTOBUSTOOLS_REF.out.index
                input[2] = KALLISTOBUSTOOLS_REF.out.t2g
                input[3] = KALLISTOBUSTOOLS_REF.out.cdna_t2c.ifEmpty{ [] }   // when empty the module doesn't run unless something is passed.
                input[4] = KALLISTOBUSTOOLS_REF.out.intron_t2c.ifEmpty{ [] } // when empty the module doesn't run unless something is passed.
                input[5] = "10XV3"
                input[6] = "standard"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.matrix,
                    path(process.out.count.get(0).get(1)).list().findAll { file(it.toString()).name != "run_info.json" && file(it.toString()).name != "kb_info.json" },
                    file(path(process.out.count.get(0).get(1)).list().find { file(it.toString()).name == "kb_info.json" }.toString()).readLines()[2..10],
                    file(path(process.out.count.get(0).get(1)).list().find { file(it.toString()).name == "run_info.json" }.toString()).readLines()[0..9]
                    ).match()
                }
            )
        }
    }

    test("genome.fasta + genome.gtf + '10X3' + 'standard' - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [id:'test'], // meta map
                        [
                        file(params.test_data['homo_sapiens']['10xgenomics']['cellranger']['test_10x_10k_pbmc_cmo_gex1_fastq_1_gz'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['10xgenomics']['cellranger']['test_10x_10k_pbmc_cmo_gex1_fastq_2_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = KALLISTOBUSTOOLS_REF.out.index
                input[2] = KALLISTOBUSTOOLS_REF.out.t2g
                input[3] = KALLISTOBUSTOOLS_REF.out.cdna_t2c.ifEmpty{ [] }   // when empty the module doesn't run unless something is passed.
                input[4] = KALLISTOBUSTOOLS_REF.out.intron_t2c.ifEmpty{ [] } // when empty the module doesn't run unless something is passed.
                input[5] = "10XV3"
                input[6] = "standard"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
