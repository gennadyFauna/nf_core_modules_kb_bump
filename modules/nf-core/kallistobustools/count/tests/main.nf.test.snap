{
    "genome.fasta + genome.gtf + '10X3' + 'standard' - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "cells_x_genes.mtx:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ]
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,6ec06270afe0a7572c41567160d927d9"
                ],
                "2": [
                    "cells_x_genes.mtx:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "count": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "cells_x_genes.mtx:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ]
                        ]
                    ]
                ],
                "matrix": [
                    "cells_x_genes.mtx:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,6ec06270afe0a7572c41567160d927d9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-29T23:04:14.434480274"
    },
    "genome.fasta + genome.gtf + '10X3' + 'standard'": {
        "content": [
            [
                "versions.yml:md5,6ec06270afe0a7572c41567160d927d9"
            ],
            [
                "cells_x_genes.mtx:md5,e12a45e7f7d6527f698dd9cb2e99ecd1"
            ],
            [
                "10x_version3_whitelist.txt:md5,3d36d0a4021fd292b265e2b5e72aaaf3",
                [
                    "cells_x_genes.barcodes.txt:md5,41f7adaf43b60f2f4f62d6a7073688de",
                    "cells_x_genes.genes.names.txt:md5,b29afa75be300c7f24fbd0740a66689b",
                    "cells_x_genes.genes.txt:md5,fe6d5501923867b514a0447aa4b4995f",
                    "cells_x_genes.mtx:md5,e12a45e7f7d6527f698dd9cb2e99ecd1"
                ],
                "inspect.json:md5,bafb47a58ac1bbf9be953f21c361d266",
                "matrix.ec:md5,31a4c1a3e8e0c562b12f6569ffbf5459",
                "output.bus:md5,d6fa0612a4a16eaf8a3e08bdc13ff49c",
                "output.unfiltered.bus:md5,bf899b967657f612ba864188868d58cc",
                "transcripts.txt:md5,23861cf43033e7c596e6989a88a3a373"
            ],
            [
                "    \"version\": \"0.28.2\",",
                "    \"kallisto\": {",
                "        \"path\": \"/usr/local/lib/python3.8/site-packages/kb_python/bins/linux/kallisto/kallisto\",",
                "        \"version\": \"0.50.1\"",
                "    },",
                "    \"bustools\": {",
                "        \"path\": \"/usr/local/lib/python3.8/site-packages/kb_python/bins/linux/bustools/bustools\",",
                "        \"version\": \"0.43.2\"",
                "    },"
            ],
            [
                "{",
                "\t\"n_targets\": 12,",
                "\t\"n_bootstraps\": 0,",
                "\t\"n_processed\": 10000,",
                "\t\"n_pseudoaligned\": 26,",
                "\t\"n_unique\": 26,",
                "\t\"p_pseudoaligned\": 0.3,",
                "\t\"p_unique\": 0.3,",
                "\t\"kallisto_version\": \"0.50.1\",",
                "\t\"index_version\": 13,"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-29T23:03:47.163313332"
    }
}